### ISSUE TITLE
ReorderColumns - Filtervalues are passed in reordered column index

### ISSUE THREAD

**Issue by tpreussFN** [vedmack/yadcf#519]
Hello,

I initialized my table with:
- sDom: ...,
- scrollX: true,
- "bAutoWidth": false,
- sPaginationType: "full_numbers",
- bSortCellsTop: true,
- bJQueryUI: true,
- aaSorting: [[ 7, 'asc' ],[ 0, 'desc' ]],
- select: false,
- stateSave:true,
- processing: true,
- serverSide: true,
- colReorder: true,
- "initComplete": afterInit,
- "ajax":{}

together with the setup for columns including the data attribute.

Then I am initializing the filters with: yadcf.init(table,[allcols], , {filters_tr_index: 1}):

Everything is working fine until the moment I want to search in a column which got reordered beforehand. 

The searchquery to the backend puts the values in the column where the column is currently. This means that for example when I move column 2 to 0 also the column[2][search][value] swaps to column[0][search][value]. 

This breaks our backend since the backend doesn't know that the columns have switched.

I am using the latest version of datatables and also the latest yadcf beta version but I also tried it out with the current release.

Thank you for checking!

Cheers!

**Comment by vedmack** [comment]
Hi
For start please check without `{filters_tr_index: 1}`

**Comment by tpreussFN** [comment]
Hi,
I removed the filters_tr_index which makes it now kinda unpossible to test because the asc/desc sorting overlapps now. I moved a column and searched in the global search field. I saw in the parameters that the number changed from 

columns[0][data]: cost
columns[0][name]: 
columns[0][searchable]: true
columns[0][orderable]: true
columns[0][search][value]: 
columns[0][search][regex]: false

to 

columns[6][data]: cost
columns[6][name]: 
columns[6][searchable]: true
columns[6][orderable]: true
columns[6][search][value]: 
columns[6][search][regex]: false

which makes it for the backend difficult to find out how to filter the value. Any ideas? I can adjust the backend to filter the "data" attribute first but I have the feeling it can be much better.

**Comment by vedmack** [comment]
Please take a look at the [showcase server side example](http://yadcf-showcase.appspot.com/server_side_source.html) it support column reordering, and it does work just fine, inspect the data that is being transferred over the network and try to find what different on my implementation, because if i remember it right it should work out of the box

**Comment by tpreussFN** [comment]
Hi!

So I checked out your serverside example which basically has the same "issue". As long as you dont rely on the exact index for your backend it is working perfectly fine.

In my case I had to edit the passed values a little bit. Therefore if the index changed the application threw an error because the now passed value wasn't able to be edited in the desired way. I changed the way the backend reads out the values and now everything is working fine.

Cheers!
